Add EmailDeliveryFailure view

NikkiWines opened the pull request:
### Details
Adds a new "email delivery failed" view to newDot for when users are blocked because AWS / Mailgun haven't been able to deliver emails to them.

### Fixed Issues
$ https://github.com/Expensify/Expensify/issues/286767
PROPOSAL: N/A

### Tests
1. Login into your account on oldDot and add the email delivery failed NVP by pasting the following in the JS console: `NVP.set('expensify_emailDeliveryFailure', {'lastSMTPErrorDatetime': moment.now().addDays(1).toString()});`
2. In an incognito window, try to log into your account in newDot 
4. Confirm you see the following view: 
     <img width="391" alt="image" src="https://github.com/Expensify/App/assets/16747822/8713f11d-f669-41e5-b9f5-ef0b986a3f84">
5. Click the `here` link and confirm it redirects to https://community.expensify.com/discussion/5651/deep-dive-best-practices-when-youre-running-into-trouble-receiving-emails-from-expensify/p1?new=1
6. Click `Back` and confirm it takes you back to login view to enter an email/sms number 
7. In your original window run `NVP.set('expensify_emailDeliveryFailure', null);` in the JS console to unblock your account

- [x] Verify that no errors appear in the JS console

### Offline tests
1. Login into your account on oldDot and add the email delivery failed NVP by pasting the following in the JS console: `NVP.set('expensify_emailDeliveryFailure', {'lastSMTPErrorDatetime': moment.now().addDays(1).toString()});`
2. In an incognito window, try to log into your account in newDot 
3. Disable your network connection
4. Confirm you see the following view: 
<img width="484" alt="Screenshot 2023-06-06 at 14 58 53" src="https://github.com/Expensify/App/assets/16747822/d2c33c07-ddd9-41da-a75a-4a126f2435f5">


### QA Steps
1. Login into your account on oldDot and add the email delivery failed NVP by pasting the following in the JS console: `NVP.set('expensify_emailDeliveryFailure', {'lastSMTPErrorDatetime': moment.now().addDays(1).toString()});`
2. In an incognito window, try to log into your account in newDot 
4. Confirm you see the following view: 
     <img width="391" alt="image" src="https://github.com/Expensify/App/assets/16747822/653963f7-5fcd-43e5-b964-e43cec6e22e9">
5. Click the `here` link and confirm it redirects to https://community.expensify.com/discussion/5651/deep-dive-best-practices-when-youre-running-into-trouble-receiving-emails-from-expensify/p1?new=1
6. Click `Back` and confirm it takes you back to login view to enter an email/sms number 
7. In your original window run `NVP.set('expensify_emailDeliveryFailure', null);` in the JS console to unblock your account

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

![image](https://github.com/Expensify/App/assets/16747822/0537e3e0-465e-41c8-865f-8e5e3a640c7e)

</details>

<details>
<summary>Mobile Web - Chrome</summary>

![Screenshot_1685749319](https://github.com/Expensify/App/assets/16747822/7aca1921-b7d9-4a9b-8b32-b9e747dd170d)

</details>

<details>
<summary>Mobile Web - Safari</summary>

![Simulator Screen Shot - iPhone 14 - 2023-06-02 at 16 39 49](https://github.com/Expensify/App/assets/16747822/1c2459ef-0b59-4355-8ac8-b52e14403692)

</details>

<details>
<summary>Desktop</summary>

![Screenshot 2023-06-02 at 16 17 20](https://github.com/Expensify/App/assets/16747822/7e3fe3cd-7406-4b1c-afb7-35efecf7e22e)

</details>

<details>
<summary>iOS</summary>

![Simulator Screen Shot - iPhone 14 - 2023-06-02 at 16 38 59](https://github.com/Expensify/App/assets/16747822/5854da75-0f02-42f8-a567-03f377b6b0f1)

</details>

<details>
<summary>Android</summary>

<img width="454" alt="image" src="https://github.com/Expensify/App/assets/16747822/d988d3b4-cc9b-4b97-bc17-3da2301a2747">

</details>

NikkiWines committed:
add copy for email delivery failure

NikkiWines committed:
add email delivery page to signin logic

NikkiWines committed:
style and consistentcy

NikkiWines committed:
add emailDeliveryFailurePage

NikkiWines committed:
Merge branch 'main' of https://github.com/Expensify/App into nikki-emaildeliveryissue

NikkiWines committed:
Merge branch 'main' of https://github.com/Expensify/App into nikki-emaildeliveryissue

NikkiWines committed:
update copy

NikkiWines committed:
update styling for emaildeliveryfailure view

NikkiWines committed:
ensure we show the correct header

NikkiWines committed:
jsx style

NikkiWines committed:
prettier style

NikkiWines committed:
remove out of date comment

NikkiWines committed:
Merge branch 'main' of https://github.com/Expensify/App into nikki-emaildeliveryissue

NikkiWines committed:
update english copy

NikkiWines committed:
add spanish copy

NikkiWines committed:
use presablewithfeedback instead of touchable opacity

NikkiWines committed:
Merge branch 'main' of https://github.com/Expensify/App into nikki-emaildeliveryissue

NikkiWines committed:
fix imports

NikkiWines committed:
Update emailDeliveryFailure logic to match new function component style + merge main

NikkiWines committed:
function docs, prop validation, minor style

NikkiWines committed:
typo

NikkiWines committed:
prettier style

NikkiWines commented:
Some changes going on with the `SignInPage` where it was converted to a functional component and then reverted so going to hold off on updates for now

NikkiWines commented:
Looks like there's a [WIP PR](https://github.com/Expensify/App/pull/20708) to migrate that component again so holding off until that's merged

NikkiWines commented:
https://github.com/Expensify/App/pull/20708 is in review so going to continue to hold off on making changes

NikkiWines committed:
Merge branch 'main' of github.com:Expensify/App into nikki-emaildeliveryissue

NikkiWines committed:
update function declaration

Santhosh-Sellavel commented on src/pages/signin/SignInPage.js:
```diff
@@ -62,18 +66,23 @@ const defaultProps = {
-    const shouldShowResendValidationForm = hasLogin && (!isAccountValidated || didForgetPassword) && !isUnvalidatedSecondaryLogin && !canUsePasswordlessLogins;
-    const shouldShowWelcomeHeader = shouldShowLoginForm || shouldShowPasswordForm || shouldShowValidateCodeForm || isUnvalidatedSecondaryLogin;
-    const shouldShowWelcomeText = shouldShowLoginForm || shouldShowPasswordForm || shouldShowValidateCodeForm;
+    const shouldShowEmailDeliveryFailurePage = hasLogin && hasEmailDeliveryFailure;
+    const isUnvalidatedSecondaryLogin = hasLogin && !isPrimaryLogin && !isAccountValidated && !shouldShowEmailDeliveryFailurePage;
+    const shouldShowPasswordForm =
+        hasLogin && isAccountValidated && !hasPassword && !didForgetPassword && !isUnvalidatedSecondaryLogin && !canUsePasswordlessLogins && !shouldShowEmailDeliveryFailurePage;
+    const shouldShowValidateCodeForm = (hasLogin || hasValidateCode) && !isUnvalidatedSecondaryLogin && canUsePasswordlessLogins && !shouldShowEmailDeliveryFailurePage;
+    const shouldShowResendValidationForm =
+        hasLogin && (!isAccountValidated || didForgetPassword) && !isUnvalidatedSecondaryLogin && !canUsePasswordlessLogins && !shouldShowEmailDeliveryFailurePage;
+    const shouldShowWelcomeHeader = shouldShowLoginForm || shouldShowPasswordForm || shouldShowValidateCodeForm || isUnvalidatedSecondaryLogin || shouldShowEmailDeliveryFailurePage;
+    const shouldShowWelcomeText = shouldShowLoginForm || shouldShowPasswordForm || shouldShowValidateCodeForm || shouldShowEmailDeliveryFailurePage;
```
Ah, too many checks 😱

Santhosh-Sellavel commented on src/pages/signin/EmailDeliveryFailurePage.js:
```diff
@@ -0,0 +1,91 @@
+
+EmailDeliveryFailurePage.propTypes = propTypes;
+EmailDeliveryFailurePage.defaultProps = defaultProps;
+EmailDeliveryFailurePage.displayName = 'ResendValidationForm';
```
Display name is wrong

Santhosh-Sellavel commented:
@NikkiWines did a quick test seems to work well. I get to see the failure view.

<img width="751" alt="Screenshot 2023-07-05 at 12 29 08 AM" src="https://github.com/Expensify/App/assets/85645967/e7bd576e-0abc-40f4-af8a-e9de53727109">


#### There are a couple of related prop errors/warnings please fix that, thanks

<img width="684" alt="Screenshot 2023-07-05 at 12 29 23 AM" src="https://github.com/Expensify/App/assets/85645967/26599b6a-7d58-4a62-af2d-60abd8837318">

<img width="680" alt="Screenshot 2023-07-05 at 12 29 38 AM" src="https://github.com/Expensify/App/assets/85645967/2d2deb78-a530-421a-9105-e526f13524cf">

NikkiWines committed:
update display name, getRenderOptions(), and welcomeText for email delivery failed view

NikkiWines committed:
Merge branch 'main' of github.com:Expensify/App into nikki-emaildeliveryissue

NikkiWines committed:
prettier style

NikkiWines commented:
thanks for the review @Santhosh-Sellavel, updated!

Santhosh-Sellavel commented:
@NikkiWines Seems the warning still occurs please check the video below

https://github.com/Expensify/App/assets/85645967/a7d20f56-06cc-47cb-9198-8a215b10203e

NikkiWines committed:
Merge branch 'main' of github.com:Expensify/App into nikki-emaildeliveryissue

NikkiWines committed:
re-add missing param

ensure that we're always setting welcomeHeader and welcomeText for cases where they might be used

NikkiWines committed:
prettier

NikkiWines commented on src/pages/signin/SignInPage.js:
```diff
@@ -112,11 +122,12 @@ function SignInPage({credentials, account}) {
     let welcomeHeader;
     let welcomeText;
-    if (shouldShowValidateCodeForm) {
+    if (shouldShowValidateCodeForm || shouldShowResendValidationForm) {
```
We were somehow returning true for `shouldShowResendValidationForm` (which I think should probably be removed) but never had any logic for setting the `welcomeHeader` or `welcomeText` for that view. Now we do 😅

NikkiWines commented:
Ah, sorry about that @Santhosh-Sellavel, should be good now

Santhosh-Sellavel commented on src/pages/signin/EmailDeliveryFailurePage.js:
```diff
@@ -0,0 +1,91 @@
+EmailDeliveryFailurePage.displayName = 'EmailDeliveryFailurePage';
+
+export default compose(
+    withLocalize,
```
Can we use `useLocalizeHook` here instead of `withLocalize` HOC

NikkiWines replied:
Yep, good catch!

NikkiWines committed:
Merge branch 'main' of github.com:Expensify/App into nikki-emaildeliveryissue

NikkiWines committed:
use useLocalize instead of withLocalize

NikkiWines commented:
Updated once more 🙇

NikkiWines committed:
don't use compose for single export

NikkiWines committed:
remove unused import

Santhosh-Sellavel commented:
@NikkiWines Keyboard Doesn't disappear automatically on iOS alone.

https://github.com/Expensify/App/assets/85645967/d7de6228-87b1-496f-9c7f-e9cf5b261bcd

Santhosh-Sellavel commented:
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android / native
    - [x] Android / Chrome
    - [ ] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web & Desktop </summary>

https://github.com/Expensify/App/assets/85645967/66d1e0a4-dc06-4959-bfc6-c2d40d63703a

</details>

<details>
<summary>Mobile Web - Chrome & Mobile Web - Safari</summary>

https://github.com/Expensify/App/assets/85645967/89585d8c-8c9a-4bbb-b604-6b659f0fe03b

</details>

<details>
<summary>iOS & Android</summary>

https://github.com/Expensify/App/assets/85645967/777e720e-bf74-40ff-a433-82437729b084

</details>

Santhosh-Sellavel reviewed (commented):
Other than [this](https://github.com/Expensify/App/pull/19508#issuecomment-1630984926) LGTM, tests well!

NikkiWines committed:
dismiss keyboard on email delivery failure page

NikkiWines commented:
@Santhosh-Sellavel added logic to dismiss the keyboard

NikkiWines committed:
lint errors

Santhosh-Sellavel commented on src/pages/signin/EmailDeliveryFailurePage.js:
```diff
@@ -0,0 +1,96 @@
+    const {translate} = useLocalize();
+    const login = Str.isSMSLogin(props.credentials.login) ? Str.removeSMSDomain(props.credentials.login) : props.credentials.login;
+
+    useEffect(() => {
+        if (!isKeyboardShown) {
+            return;
+        }
+        Keyboard.dismiss();
+    }, [isKeyboardShown]);
```
Let's add a comment why this was added, as it look unnecessary here

NikkiWines replied:
Done!

NikkiWines committed:
add clarifying comments

Santhosh-Sellavel reviewed (approved):
LGTM tests well!

PauloGasparSv reviewed (approved):
LGTM

<details>
<summary>Web</summary>

<img width="1462" alt="image" src="https://github.com/Expensify/App/assets/6564265/15aaebab-4d0e-4772-9247-8ab8666f71af">
<img width="677" alt="image" src="https://github.com/Expensify/App/assets/6564265/611a4d47-6a04-4554-bb25-0bae2e437bd2">

</details>

<details>
<summary>Mobile Web - Chrome</summary>

Skipped, having trouble with the android emulator. Stuck on the loading screen on chrome mobile

</details>

<details>
<summary>Mobile Web - Safari</summary>

<img width="381" alt="image" src="https://github.com/Expensify/App/assets/6564265/80b7dbe1-e983-46b8-9186-787734543492">

</details>

<details>
<summary>Desktop</summary>
<img width="1186" alt="image" src="https://github.com/Expensify/App/assets/6564265/34bb0673-3297-4e3a-ac7e-61a6ad5ae09e">

<img width="855" alt="image" src="https://github.com/Expensify/App/assets/6564265/027c7362-27ab-476c-848b-677d3a6048f0">

</details>

<details>
<summary>iOS</summary>

<img width="412" alt="image" src="https://github.com/Expensify/App/assets/6564265/f792a88b-9cbe-44bf-80a4-810b6c1cc880">

</details>

<details>
<summary>Android</summary>

<img width="426" alt="image" src="https://github.com/Expensify/App/assets/6564265/dccbc975-16a2-41f7-b50e-e5e674946181">

</details>

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/5538077342) to staging by https://github.com/PauloGasparSv in version: 1.3.40-0 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|success ✅ 
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅ 
🕸 web 🕸|success ✅

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/5555174239) to production by https://github.com/Julesssss in version: 1.3.40-5 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|success ✅ 
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅ 
🕸 web 🕸|success ✅

mountiny commented:
There is a regression in the back end so going to fix that

NikkiWines commented:
@Santhosh-Sellavel issue created! Sorry about that - https://github.com/Expensify/App/issues/22910
